### PR TITLE
[ISSUE #4175] Prevent update topic on slave

### PR DESCRIPTION
**Pull request by HScarb**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

#4175 

## Brief changelog

When modify topic from slave, return a error response

## Verifying this change

image
image
image
image

### CODE REVIEW

**Review by yuz10 — APPROVED** [review]

**Comment by yuz10** [comment]
Create subscription group on slave have the same problem, would you correct that?

**Review by RongtongJin — APPROVED** [review]

**Comment by HScarb** [comment]
> Create subscription group on slave have the same problem, would you correct that?

I would like to correct that

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java [hunk 1]
```diff
@@ -1636,4 +1649,14 @@ private MessageExtBrokerInner toMessageExtBrokerInner(MessageExt msgExt) {
         inner.setWaitStoreMsgOK(false);
         return inner;
     }
+
+    private boolean validateSlave(RemotingCommand response) {
```
Comment (ltamber) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:1653]: I think the method name `validateSlave` meaning only check but not modify, maybe we should use a name mean `adjust response if slave`

**Comment by ni-ze** [comment]
@HScarb Travis CI failed.

**Review by panzhi33 — APPROVED** [review]

**Review by duhenglucky — APPROVED** [review]

**Comment by absolute8511** [comment]
Does it need to valid delete action on slave? It may cause the topic remain undeleted on slave if we do this. And by the way why it is not merge into the newest 5.1?

**Pull request merged**